source/sundials/sunlinsol_wrapper.cc

```diff
@@ -254,10 +258,15 @@ namespace SUNDIALS
     }
 
 
-
     template <typename VectorType>
     int
-    arkode_linsol_set_a_times(SUNLinearSolver LS, void *A_data, ATimesFn ATimes)
+    arkode_linsol_set_a_times(SUNLinearSolver LS,
+                              void           *A_data,
+#  if DEAL_II_SUNDIALS_VERSION_GTE(6, 0, 0)
+                              SUNATimesFn ATimes)
+#  else
+                              ATimesFn        ATimes)
+#  endif
```

Comment:
This is rather ugly. I could also introduce an alias for these types in `dealii/sundials/sundialfs_types.h` if you'd like.

Reply:
I think it looks nicer if you move the closing parenthesis out of the `#if`...`#endif` block. Other than that, I think this is just fine with me.

Reply:
I moved it out of the block.

Reply:
@masterleinad What do you think about adding an alias? You introduced the one for `realtype` in #16271.

Reply:
Sure. 🙂